roles/kubernetes-apps/metallb/defaults/main.yml

```diff
@@ -19,4 +19,4 @@ metallb_speaker_tolerations:
     operator: Exists
 metallb_controller_tolerations: []
 metallb_pool_name: "loadbalanced"
-matallb_auto_assign: true
+metallb_auto_assign: true
```

Comment:
I wonder if anyone is actually using the typo'ed variable name. Carrying some compatibility code due to a typo in the past seems like a lot of effort.

Reply:
Yeah, this kind of thing is always maintenance burden for developers.
At least, it would be necessary to take 1-2 versions for warning the option is renamed from the original one for users as the deprecation period.
Then we can remove the original one after that.

Reply:
Well this can be discussed, but a warning in the release note might be enough, or we can add an ansible check and message indeed

Reply:
I updated this PR with the solution proposed by @oomichi , but I can still modify it if this discussion ends up identifying another solution.

Reply:
> we can add an ansible check and message indeed

@floryut Thanks, that is a good idea for users I think.
@orange-llajeanne Could you update this pull request again based on
https://github.com/kubernetes-sigs/kubespray/blob/4726a110fc6f6428a1caf04168ee9ded198a833f/roles/network_plugin/calico/tasks/check.yml#L2-L8 
as a reference?